src/main/java/org/jitsi/jicofo/BridgeSelector.java

```diff
@@ -1002,9 +1017,20 @@ private Bridge findFirst(
             List<Bridge> selectFrom, List<Bridge> order)
         {
             return order.stream()
+                .filter(not(Bridge::isInSurvivalMode))
                 .filter(selectFrom::contains)
                 .findFirst()
-                .orElse(selectFrom.get(0));
+                .orElse(selectFrom.stream()
+                    .filter(not(Bridge::isInSurvivalMode))
+                    .findFirst()
+                    .orElse(order.stream()
+                        .filter(selectFrom::contains)
+                        .findFirst()
+                        .orElse(selectFrom.get(0))));
         }
     }
+
+    public static <T> Predicate<T> not(Predicate<T> t) {
```

Comment:
This is cool!

Reply:
more like legacy code if you think that Java11 already has this baked in :/